src/Molecule.py

```diff
@@ -166,28 +169,46 @@ def filter_properties(self):
 class R_group(Molecule):
     """Name of R group is of the form 'Axy' or 'Bxy' e.g. A01 etc.
     Number corresponds to whether it is an R1 or R2 group"""
-    def __init__(self, name, number):
+
+    def __init__(self, name, number=None):
         self.name = name
+        if number is None:
+            if name[0] == 'A':
+                number = 1
+            elif name[0] == 'B':
+                number = 2
+            else:
+                raise ValueError('R group ID not valid')
         self.number = number
         mol_smiles = self.extract_smilefromcsv()
         super().__init__(mol_smiles)
 
     def extract_smilefromcsv(self):
         """Extracts the SMILE for the R group """
-
-        if self.number == 1:
-            rgroup_smiles = csv_file[csv_file['atag'] ==
-                                     self.name]['R1'].iloc[0]
-        if self.number == 2:
-            rgroup_smiles = csv_file[csv_file['btag'] ==
-                                     self.name]['R2'].iloc[0]
+        try:
+            if self.number == 1:
+                rgroup_smiles = csv_file[csv_file['atag'] ==
+                                         self.name]['R1'].iloc[0]
+            if self.number == 2:
+                rgroup_smiles = csv_file[csv_file['btag'] ==
+                                         self.name]['R2'].iloc[0]
+        except ValueError('Invalid R Group ID'):
+            raise ValueError('Invalid R Group ID')
         return(rgroup_smiles)
 
+    def add_r_group(self, base_molecule):
```

Comment:
Should you explain what this function does (i.e """ """ - sorry I can't remember what this commenting is called)?